### PR TITLE
HPCC-32424 ECL Watch v9 Source Files default sorting

### PR DESCRIPTION
**Pull request by jeclrsg**
changes the default sorting of files listed on the Source Files tab of the Workunit Details page, such that subfiles immediately follow their respective superfiles

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/hooks/workunit.ts [hunk 1]
```diff
@@ -132,6 +132,29 @@ export function useWorkunitSourceFiles(wuid: string): [SourceFile[], Workunit, W
...
+        });
+
+        parents.forEach(parent => {
+            sortedFiles.push(parent);
+            const relatedChildren = children.filter(child => child.__hpcc_parentName === parent.Name);
+            sortedFiles.push(...relatedChildren);
+        });
+
+        return sortedFiles;
+    }, []);
```
Comment (GordonSmith) [inline esp/src/src-react/hooks/workunit.ts:156]: Will this work for nesting > 2 levels deep?
Is it not possible to do it within a single sort call?

**Review by GordonSmith — CHANGES_REQUESTED** [review]
One question

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/hooks/workunit.ts [hunk 2]
```diff
@@ -132,6 +132,21 @@ export function useWorkunitSourceFiles(wuid: string): [SourceFile[], Workunit, W
...
+        const sortedFiles = [];
+        const temp = files.sort((a, b) => a.Name.localeCompare(b.Name));
+
+        temp.filter(item => item.__hpcc_parentName === "").forEach(parent => {
+            sortedFiles.push(parent);
+            const relatedChildren = temp.filter(child => child.__hpcc_parentName === parent.Name);
+            sortedFiles.push(...relatedChildren);
+        });
+
+        return sortedFiles;
```
[the remark this answers is not in the document]
Comment (jeclrsg) [reply esp/src/src-react/hooks/workunit.ts:147]: Doesn't seem to work the same; children don't immediately follow their parents with this method

[the remark this answers is not in the document]
Comment (GordonSmith) [reply esp/src/src-react/hooks/workunit.ts:147]: In the raw data, or on the page?

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**